### ISSUE TITLE
[shared_preferences] Support sharing preferences between Flutter & Native

### ISSUE THREAD

**Issue by btrautmann** [flutter/flutter#87658]
## Use case

Our team has a use case that would be made much easier (nearly trivial) by having the ability to persist data across app processes that can be shared between Flutter and native. In other words, whether the Flutter side writes this data or the native side does, both sides should be able to read this data.

I can't speak to iOS as my background is primarily in Android, but the first barrier to this is that the file name used to open the `SharedPreferences` file is [static](https://github.com/flutter/plugins/blob/master/packages/shared_preferences/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/MethodCallHandlerImpl.java#L37) and can't be configured. Additionally, each value is [stored with a `.flutter` prefix](https://github.com/flutter/plugins/blob/master/packages/shared_preferences/shared_preferences/lib/shared_preferences.dart#L147) and then when accessed, [only values stored by Flutter are fetched](https://github.com/flutter/plugins/blob/master/packages/shared_preferences/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/MethodCallHandlerImpl.java#L187). 

The information above leads me to believe that this choice to _not_ allow sharing was done intentionally, and I'm wondering if there still remains good reason not to allow both sides of the plugin to work together.

## Proposal

In a perfect world, users of the `shared_preferences` plugin would be able to specify the information necessary to support sharing. This would not need to be enabled by default, but should one want to, they should be able to indicate that they want to read values written by native and vice versa.

I'd be open to contributing myself, but wanted to first check to see if there was any point in trying.

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/52544

(The use case is slightly different, but the functionality to enable it should be the same.)

**Comment by stuartmorgan** [comment]
There are at least two PRs intended to solve this in various states of completeness that you'd want to evaluate before starting a new PR.)
